runtime/commonMain/src/kotlinx/serialization/internal/SerialClassDescImpl.kt

```diff
@@ -81,21 +94,25 @@ open class SerialClassDescImpl @JvmOverloads constructor(
     override fun equals(other: Any?): Boolean {
         if (this === other) return true
         if (other !is SerialClassDescImpl) return false
-        if (name != other.name) return false
+        if (serialName != other.serialName) return false
         if (elementDescriptors() != other.elementDescriptors()) return false
         return true
     }
 
     override fun hashCode(): Int {
-        var result = name.hashCode()
+        var result = serialName.hashCode()
         result = 31 * result + elementDescriptors().hashCode()
         return result
     }
 
     override fun toString(): String {
-        return indices.entries.joinToString(", ", "$name(", ")") { it.key + ": " + getElementDescriptor(it.value).name }
+        return indices.entries.joinToString(", ", "$serialName(", ")") { it.key + ": " + getElementDescriptor(it.value).serialName }
     }
+}
 
-    private class MissingDescriptorException(index: Int, origin: SerialDescriptor) :
-        SerializationException("Element descriptor at index $index has not been found in ${origin.name}")
+internal fun SerialDescriptor(name: String, kind: SerialKind): SerialDescriptor {
```

Comment:
'Find usages' in IDEA shows that documentation links like `[SerialDescriptor]` were linked here, not to interface. Since it is internal anyway, you can choose another name